File: src/main/java/com/sat/member/infrastructure/repository/MemberRepository.java
package com.sat.member.infrastructure.repository;

import com.sat.member.domain.Member;
import com.sat.member.domain.MemberId;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface MemberRepository extends JpaRepository<Member, MemberId> {
    Optional<Member> findByName_Value(String name);

    Optional<Member> findById(MemberId id);
}

File: src/main/kotlin/com/sat/user/command/domain/member/MemberRepository.kt
package com.sat.user.command.domain.member

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface MemberRepository : JpaRepository<Member, Long> {
    @Query(
        """
            select m from Member m
            join fetch m.role
            where m.email = :email
        """
    )
    fun findByEmail(@Param("email") email: String): Member?
}
